### ISSUE TITLE
Fix twisted words irc.py documentation string

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4261]
|<img alt="Mike.lifeguard's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| Mike.lifeguard reported|
|-|-|
|Trac ID|trac#4261|
|Type|defect|
|Created|2010-02-08 02:03:18Z|
|Branch|https://github.com/twisted/twisted/tree/irc-wtf-4261|

At http://twistedmatrix.com/trac/browser/trunk/twisted/words/protocols/irc.py line 489, H an G mean "here" and "gone"

<details><summary>Searchable metadata</summary>

```
trac-id__4261 4261
type__defect defect
reporter__Mike_lifeguard Mike.lifeguard
priority__low low
milestone__ 
branch__branches_irc_wtf_4261 branches/irc-wtf-4261
branch_author__jesstess jesstess
status__closed closed
resolution__fixed fixed
component__words words
keywords__ 
time__1265594598000000 1265594598000000
changetime__1265997425000000 1265997425000000
version__None None
owner__ 
cc__jesstess cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess set owner to @jesstess|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="Mike.lifeguard's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_2"></a>|Mike.lifeguard commented|
|-|-|

Note also that typical /who replies contain an additional character showing if the user is voiced (+) opped (@) or something (some networks/ircds have halfops for example). This is also missing.

For example:
#wikimedia Mike wikimedia/mikelifeguard niven.freenode.net Mike_lifeguard H@ :0 PGP key: 0xB2DD0047F0DA287B

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_3"></a>|@jesstess commented|
|-|-|

(In [28125]) Branching to 'irc-wtf-4261'

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_4"></a>|@jesstess commented|
|-|-|

(In [28126]) Change the wtf comment in the docstring to what H and G stand
for. Also fix linewrapping and @param indention for the rest of class
IRC.

refs [#4261](https://github.com/twisted/twisted/issues/4261)

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_5"></a>|@jesstess removed owner|
|-|-|

Does twisted.words.protocols.IRC support privilege flags like voiced or opped? The assert in who() suggests no. I'll put the doc change up for review and can make an enhancement ticket for that if need be.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra set owner to @jesstess|
|-|-|

I'm not such a fan of the 'Implement me!' line (that also contains trailing whitespace). Can you make this more formal?

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_7"></a>|@jesstess commented|
|-|-|

(In [28155]) Make irc_unknown docstring a little more informative.

refs [#4261](https://github.com/twisted/twisted/issues/4261)

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun set owner to @jesstess|
|-|-|

I like the new docstring for `irc_unknown`.  It's much better than the previous one.  I think we should probably leave a hint to readers that this is an intended customization point for subclasses, though.  How about sticking something simple like "Subclasses should override this method." at the end?

For the `notice` docstring, I don't think we need to \ "s anymore (it was a workaround for an emacs python-mode bug years ago, iirc).

The rest looks good.  Please merge once the above two points are dealt with.  Thanks.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_10"></a>|@jesstess commented|
|-|-|

(In [28178]) Be clear in irc_unknown about the fact that you should override the
method, and un-escape quotes in the file as that's not necessary
anymore.

refs [#4261](https://github.com/twisted/twisted/issues/4261)

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_11"></a>|@jesstess set status to `closed`|
|-|-|

(In [28180]) Merge irc-wtf-4261

Author: jesstess
Reviewers: thijs, exarkun
Fixes: [#4261](https://github.com/twisted/twisted/issues/4261)

Explain the flags associated with a user in a channel in the IRC.who
docstring.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
